src/components/standalone-link/standalone-link.module.css

```diff
@@ -58,22 +63,25 @@
 ***
 */
 
-.color-primary {
+.primary {
 	color: var(--token-color-foreground-action);
 
-	&:active {
+	/* only change color on active for <a href> */
+	&[href]:active {
 		color: var(--token-color-foreground-action-active);
 	}
 
-	&:hover {
+	/* only change color on active for <a href> */
+	&[href]:hover {
```

Comment:
💭 For clarity here, this means the `:hover` color change effect will no longer apply to `StandaloneLinkContents`, but the underline-on-hover will still apply, right? I think we might want to retain the `:hover` for now given that's the behaviour upstream 🤔

Reply:
Nope! The underline-on-hover only happens in the `.root` class, which is only applied to `StandaloneLink`'s `<a>`. that's not super clear from just reading the CSS though. I'm going to rename some classes!

The purpose of this `href` specificity check is to make it so that `StandaloneLinkContents` does not change color on hover. This is because the component itself is just plain text, and not interactive. If color _does_ need to change on hover/active of an element `StandaloneLinkContents` is inside of, it can easily be layered on top of the base CSS for `StandaloneLinkContents`.

Since `StandaloneLinkContents` is not used anywhere on its own but the Swingset page, this separation shouldn't have any downstream affects.

Reply:
I spun up this branch here locally and was poking around here, and here's a (very sloppy) commit which illustrates my exploration here.  I believe the changes I make (which remove the inherit prop) I believe still yields a similar result.

https://github.com/hashicorp/dev-portal/commit/f3437e3180a7da744382117cbf738eb1d4dc6af1

I'll have to noodle on your approach more tomorrow.

Reply:
@BrandonRomano unfortunately it doesn't yield the same result, because the active/hover color style changes only apply to `.standaloneLink`. So when both `.standaloneLink` and `.standaloneLinkContents` have the same color class applied, `.standaloneLink`'s color changes on active/hover, but `.standaloneLinkContents`'s color stays the same and overwrites the color `.standaloneLink` would otherwise pass down for `.standaloneLinkContents` to inherit.

Here's how I tested:

- [ ] Go to the `StandaloneLink` Swingset page for your branch: https://dev-portal-git-brstandalone-link-diff-hashicorp.vercel.app/swingset/components/standalonelink
- [ ] Scroll down to one of the `StandaloneLink` examples
- [ ] Inspect the `a.standaloneLink` element of one of the examples
- [ ] Note what the `color` is set to

<img width="1647" alt="CleanShot 2023-04-06 at 21 32 36@2x" src="https://user-images.githubusercontent.com/43934258/230522415-7a4daaa6-c236-4d1e-8229-8fa5ee9dbc38.png">

- [ ] Inspect the `div.standaloneLinkContents` element directly inside of the `a.standaloneLink` element
- [ ] Note what the `color` is set to

<img width="1647" alt="CleanShot 2023-04-06 at 21 34 11@2x" src="https://user-images.githubusercontent.com/43934258/230522551-ad114138-9e50-47ff-bc5a-d5fbac8e479b.png">

- [ ] Inspect the `a.standaloneLink` element again
- [ ] Force the `:hover` state on the element using devtools
- [ ] Note what the `color` is now set to (it's correctly updating based on the color classname)

<img width="1645" alt="CleanShot 2023-04-06 at 21 35 23@2x" src="https://user-images.githubusercontent.com/43934258/230522703-3dc54de1-49cb-4c89-95f6-9754d01d511e.png">

- [ ] With the `:hover` state still forced on `a.standaloneLink`, inspect the `div.standaloneLinkContents` element directly inside of the `a.standaloneLink` element
- [ ] Note what the `color` is set to (it's still the base color, and not matching `a.standaloneLink`)

<img width="1645" alt="CleanShot 2023-04-06 at 21 40 10@2x" src="https://user-images.githubusercontent.com/43934258/230523211-0a655bc0-f796-47be-ae8c-6f0439a37445.png">

Reply:
Thank you for walking me through this here!